src/main/java/filesystem/NativeFileSystem.java

```diff
@@ -0,0 +1,38 @@
+package filesystem;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class NativeFileSystem {
+
+    private Path rootDir;
+
+    public NativeFileSystem(String rootDir) {
+        this.rootDir = Paths.get(rootDir);
+    }
+
+    public boolean fileExists(String path) {
+        return Files.exists(resolveRoot(path));
+    }
+
+    public InputStream readFile(String path) throws IOException {
+        return Files.newInputStream(resolveRoot(path));
+    }
+
+    public void writeFile(String destinationPath, InputStream source) throws IOException {
+        Files.copy(source, resolveRoot(destinationPath));
```

Comment:
Most of these system manipulations can raise an exception, you currently throw up these exception up to the _controller_ is that on purpose? It mean that currently if something go wrong, the only clue that you will have about what happened it the code + error message that you picked up a this very top level

Reply:
@fabientownsend I haven't quite thought that far yet but I was thinking of catching them in the relevant `CommandRegistry` handles so I can indicate to the client that something went wrong with them. What were you thinking as an alternative?